spark/src/test/java/org/opensearch/sql/spark/asyncquery/model/SparkSubmitParametersTest.java

```diff
@@ -30,8 +30,17 @@ public void testBuildWithExtraParameters() {
 
   @Test
   public void testBuildQueryString() {
-    String query = "SHOW tables LIKE \"%\";";
-    String params = SparkSubmitParameters.Builder.builder().query(query).build().toString();
-    assertTrue(params.contains(query));
+    String rawQuery = "SHOW tables LIKE \"%\";";
+    String expectedQueryInParams = "\"SHOW tables LIKE \\\"%\\\";\"";
+    String params = SparkSubmitParameters.Builder.builder().query(rawQuery).build().toString();
+    assertTrue(params.contains(expectedQueryInParams));
+  }
+
+  @Test
+  public void testBuildQueryStringNestedQuote() {
+    String rawQuery = "SELECT '\"1\"'";
+    String expectedQueryInParams = "\"SELECT '\\\"1\\\"'\"";
+    String params = SparkSubmitParameters.Builder.builder().query(rawQuery).build().toString();
+    assertTrue(params.contains(expectedQueryInParams));
```

Comment:
can we have more test for specical chars, like spark/src/main/java/org/opensearch/sql/spark/dispatcher/model/IndexQueryDetails.java

Reply:
I didn't add those characters because they won't be escaped, so it makes no difference.
I can add them, but without a actual EMR-S to IT test startJobRun, this unit test itself won't do much.
Updated PR description to include the end to end test for this.

Reply:
public static String escapeJava(String str)
Escapes the characters in a String using Java String rules.

Deals correctly with quotes and control-chars (tab, backslash, cr, ff, etc.)

So a tab becomes the characters '\\' and 't'.

https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringEscapeUtils.html#escapeJava(java.lang.String)

Can we add more test?